pennylane_lightning/core/src/utils/Util.hpp

```diff
@@ -533,4 +533,29 @@ auto kronProd(const std::vector<T> &diagA, const std::vector<T> &diagB)
 
     return result;
 }
+
+/**
+ * @brief Check if a matrix is a Hermitian matrix.
+ *
+ * @tparam T Data type.
+ *
+ * @param n Number of columns.
+ * @param lda Number of rows.
+ * @param mat A matrix to be checked.
+ *
+ * @return is_Hermitian Is the matrix a Hermitian matrix or not.
+ */
+template <typename T>
+bool is_Hermitian(size_t n, size_t lda,
+                  const std::vector<std::complex<T>> &mat) {
+    for (size_t i = 0; i < n; i++) {
```

Comment:
you can make the check parallel using OMP parallel-for ;) not required for this PR though

Reply:
Done!